### DISCUSSION TITLE
invalid soundfiles(.wav)

### DISCUSSION THREAD

**Discussion by deerestFarther** [voicepaw/so-vits-svc-fork#510] [Q&A]
WARNING  [13:52:19] Failed to load dataset_raw/huer_01/vocals/4_000dKydolx07mcAYvv3i01040200k7DU0k010_(Vocals)_2.wav due to Invalid file:                                               preprocess_resample.py:67
                    PosixPath('dataset_raw/huer_01/vocals/4_000dKydolx07mcAYvv3i01040200k7DU0k010_(Vocals)_2.wav') 

System architecture: I'm trying out this repo on an arm64 M1 Max chip.( I reckoned that SoC may result in better performance, who knows?) 
Problem description: At first I was on my way to the training process, then some tersor-outofboundaries exceptions popped up, I debugged it and found out it was because that I was using readaudio to read audio files instead of PySoundFile. So with some efforts, I reinstalled PySoundFile, and then the warning from above happened.(QAQ)

Efforts in Vein: I've tried renaming those wav files and made sure that they can be open by my system, but still not helpful. There must be something I'm missing, what am I supposed to do next?

**Comment by ADR3-N** [comment]
I'm no professional when it comes to doing this, but I had to run almost all of the commands at least twice before I got up and running.

There's a command listed
svc pre-resample

If you have already tried this after installing pysound/pytorch, etc, please ignore, but I hope it helps. Please keep us updated

**Reply by ADR3-N** [reply]
You may also try troubleshooting with ChatGPT. This helped me immensely, though I had to remind it multiple times I was not running on Keras.

**Comment by deerestFarther** [comment]
Dear Voicepaw Team,

I hope this email finds you well. I wanted to update you on the progress of my issue with the related packages and ChatGPT 3.5 turbo. Unfortunately, despite my efforts to reinstall the packages and seek consultation, the problem persists.

Due to some personal errands this week, I won't be able to devote more time to this until later on. However, I plan to try again with GPT 4 to see if it yields any better results. In my previous attempts, I discovered that some issues were caused by architecture incompatibility, such as the incorrect naming of a critical Unix exec file. This led to a FileNotFoundException and some dependency errors.

Moving forward, I will explore different package portfolios and work towards creating a new requirements.txt file that is more compatible with Mac users.

Thank you for your assistance thus far, and I appreciate your patience as I continue to troubleshoot this issue.

Best regards, Fred

------- Original Message -------

**Reply by ADR3-N** [reply]
Do all of your wav files have such long names?

Is this your only file, or do you have others, and no other files have problems?

**Reply by deerestFarther** [reply]
Thank you for reaching out to me. I appreciate your message.

Regarding your first question, the answer is yes but not anymore. I have previously considered this issue and attempted to resolve it by renaming my files in an ascending numerical order (e.g. 1.wav, 2.wav, 3.wav, etc.). However, despite my best efforts, I encountered the same problem with all of my wav files. As evidence, I have included a list of partial system error messages below:

WARNING [21:56:12] Failed to load dataset_raw/tw_01/170.wav due to Invalid file: PosixPath('dataset_raw/tw_01/170.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/37.wav due to Invalid file: PosixPath('dataset_raw/tw_01/37.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/218.wav due to Invalid file: PosixPath('dataset_raw/tw_01/218.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/256.wav due to Invalid file: PosixPath('dataset_raw/tw_01/256.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/122.wav due to Invalid file: PosixPath('dataset_raw/tw_01/122.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/230.wav due to Invalid file: PosixPath('dataset_raw/tw_01/230.wav') preprocess_resample.py:67
WARNING [21:56:12] Failed to load dataset_raw/tw_01/242.wav due to Invalid file: PosixPath('dataset_raw/tw_01/242.wav') preprocess_resample.py:67 WARNING [21:56:12] Failed to load dataset_raw/tw_01/136.wav due to Invalid file: PosixPath('dataset_raw/tw_01/136.wav') preprocess_resample.py:67

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by GarrettConway** [comment]
Hey @deerestFarther, I actually ran into this issue today myself due using my conda env for other stuff. Going off of the error from the first post ("Invalid file: preprocess_resample.py:67 PosixPath..."), it looks to me to be an error spit out by soundfile - a package which seems to get frequently borked on installs for me. Can you try making a fresh conda env install and uninstalling soundfile and reinstalling it from pip?


`pip uninstall soundfile`

`pip install soundfile`

Also, using a conda env setup only for this will help with debugging as versioning conflicts between common libraries will cause headaches. Note that at least for me (on windows), completely uninstalling it and reinstalling it was necessary.

**Reply by deerestFarther** [reply]
Hi there,

Thank you for your email and for sharing your experience with me. Your suggestion to uninstall and reinstall soundfile from pip was appreciated, but unfortunately, I had already tried that after encountering this issue.

I also agree with your recommendation of using a conda env setup solely for this purpose to avoid versioning conflicts between common libraries. I have already implemented this solution and am using a conda env solely for this repo from the very beginning.

Once again, thank you for your help and advice. It was greatly appreciated.

Best regards,Fred

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by abhimanyupatel** [comment]
@deerestFarther I'm running into the same issue currently. Did you ever manage to resolve this?
